codex-rs/tui/src/app.rs

```diff
@@ -224,6 +251,40 @@ impl<'a> App<'a> {
                     self.dispatch_scroll_event(scroll_delta);
                 }
                 AppEvent::CodexEvent(event) => {
+                    // Check if we're waiting for a summarization response
```

Comment:
Is there a reason this logic isn't added to `dispatch_codex_event()` instead? Much of the reason to have the `dispatch_codex_event()` helper is to keep the length of `run()` down. In a new top-level function, there will be less indenting and the code should be easier to read, as well.